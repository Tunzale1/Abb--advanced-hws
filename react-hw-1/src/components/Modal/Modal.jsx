import React from 'react';
import './Modal.scss';

class Modal extends React.Component {
  render() {
    const { header, closeButton, text, actions, onClose } = this.props;
    return (
      <div className="popup" onClick={onClose}>
        <div className="modal" onClick={(e) => e.stopPropagation()}>
          {closeButton && (
            <span className="close" onClick={onClose}>
              &times;
            </span>
          )}
          <h2>{header}</h2>
          <p>{text}</p>
          <div className="actions">{actions}</div>
        </div>
      </div>
    );
  }
}

export default Modal;
