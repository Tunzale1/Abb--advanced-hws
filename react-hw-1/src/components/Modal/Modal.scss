.popup {
    position: fixed;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    background-color: rgba(0, 0, 0, 0.8);
    display: flex;
    justify-content: center;
    align-items: center;
  }
  
  .modal {
    background-color: #fff;
    padding: 40px 100px;
    border-radius: 15px;
  }
  .close {
    position: absolute;
    top: 10px;
    right: 10px;
    font-size: 20px;
    cursor: pointer;
  }
  .modal h2 {
    margin-top: 0;
  }
  
  .actions {
    display: flex;
    justify-content: flex-end;
    margin-top: 20px;
  
    button {
      margin-left: 10px;
    }
  }
  