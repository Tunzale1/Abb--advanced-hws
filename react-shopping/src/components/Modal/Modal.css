.modal-overlay {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  background-color: rgba(83, 83, 83, 0.5764705882);
  display: flex;
  justify-content: center;
  align-items: center;
}
.modal-overlay .modal {
  background: rgb(167, 224, 255);
  width: 500px;
  margin: 0 auto;
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
}
.modal-overlay .modal .modal-content {
  color: #050505;
}
.modal-overlay .modal .modal-content .modal-header {
  padding: 0 29px;
  background: rgb(141, 215, 254);
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: 20px 35px 20px 35px;
}
.modal-overlay .modal .modal-content .modal-header .modal-title {
  font-size: 22px;
}
.modal-overlay .modal .modal-footer {
  margin-top: 30px;
  display: flex;
  justify-content: center;
  text-align: center;
  margin-bottom: 30px;
}
.modal-overlay .modal .modal-footer button {
  cursor: pointer;
  border: none;
  background: #2070d2;
  border-radius: 7px;
  font-size: 15px;
  width: 120px;
  height: 41px;
  color: #ffffff;
  margin-right: 10px;
}/*# sourceMappingURL=Modal.css.map */