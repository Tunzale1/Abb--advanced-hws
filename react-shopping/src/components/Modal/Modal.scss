.modal-overlay {
    position: fixed;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    background-color: #53535393;
    display: flex;
    justify-content: center;
    align-items: center;

    .modal {
      background: rgb(167, 224, 255);
      width: 500px;
      margin: 0 auto;
      position: absolute;
      top: 50%;
      left: 50%;
      transform: translate(-50%, -50%);
      .modal-content {
        color: #050505;
    
        .modal-header {
          padding: 0 29px;
          background: rgb(141, 215, 254);
          display: flex;
          align-items: center;
          justify-content: space-between;
          padding: 20px 35px 20px 35px;
    
          .modal-title {
            font-size: 22px;
          }
        }
        }
        .modal-footer {
          margin-top: 30px;
          display: flex;
          justify-content: center;
          text-align: center;
          margin-bottom: 30px;
          button {
            cursor: pointer;
            border: none;
            background: #2070d2;
            border-radius: 7px;
            font-size: 15px;
            width: 120px;
            height: 41px;
            color: #ffffff;
            margin-right: 10px;
          }
          
        }
      }
    }
  
  
  