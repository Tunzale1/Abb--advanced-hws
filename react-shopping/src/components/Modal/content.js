export const content = [
    {
      id: 'modal1',
      header: 'Are you want to add this item in your card?',
      closeButton: true,
      action: 'add'
    },
    {
      id: 'modal2',
      header: 'Are you want to delete this item in your card?',
      closeButton: true,
      action: 'remove'
    },
  ];
