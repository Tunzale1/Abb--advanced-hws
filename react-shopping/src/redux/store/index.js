import { createStore } from "redux";
import { reducers } from '../reducers/index';
import {applyMiddleware} from "redux";
import thunkMiddleware from 'redux-thunk'
import { composeWithDevTools } from 'redux-devtools-extension';


const store = createStore(
    reducers,
    composeWithDevTools(
        applyMiddleware(thunkMiddleware)
    )
)

export default store;