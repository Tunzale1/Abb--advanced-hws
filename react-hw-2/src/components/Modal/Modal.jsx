import React, { Component } from "react";
import PropTypes from "prop-types";
import "./Modal.scss";

class Modal extends Component {
  modalopening = (e) => {
    e.stopPropagation();
  };

  render() {
    const { header, closeButton, actions, onClose } = this.props;

    return (
      <div className="modal-overlay" onClick={onClose}>
        <div className="modal" onClick={this.modalopening}>
          <div className="modal-content">
            <div className="modal-header">
              <h3 className="modal-title">{header}</h3>
              {closeButton ? (
                <span className="close-btn" onClick={onClose}>
                  <i className="fa-regular fa-2x fa-square-minus"></i>
                </span>
              ) : (
                ""
              )}
            </div>
            <div className="modal-footer">{actions}</div>
          </div>
        </div>
      </div>
    );
  }
}

Modal.propTypes = {
  header: PropTypes.string,
  closeButton: PropTypes.bool,
  text: PropTypes.string,
  onClose: PropTypes.func
};

Modal.defaultProps = {
  header: 'Are you want to add this item in your card?',
  closeButton: true,
}

export default Modal;