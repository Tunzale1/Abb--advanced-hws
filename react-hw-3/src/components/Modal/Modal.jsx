import PropTypes from "prop-types";
import "./Modal.scss";

const Modal = ({ closeButton, modalData, actions, onClose }) => {
  const handleModalContentClick = (e) => {
    e.stopPropagation();
  };

    return (
      <div className="modal-overlay" onClick={onClose}>
        <div className="modal" onClick={handleModalContentClick}>
          <div className="modal-content">
            <div className="modal-header">
              <h3 className="modal-title">{modalData.header}</h3>
              {closeButton ? (
                <span className="close-btn" onClick={onClose}>
                  <i className="fa-regular fa-2x fa-square-minus"></i>
                </span>
              ) : (
                ""
              )}
            </div>
            <div className="modal-footer">{actions}</div>
          </div>
        </div>
      </div>
    );
  }


Modal.propTypes = {
  modalData: PropTypes.object,
  closeButton: PropTypes.bool,
  onClose: PropTypes.func
};

Modal.defaultProps = {
  closeButton: true,
  modalData: {}
}

export default Modal;