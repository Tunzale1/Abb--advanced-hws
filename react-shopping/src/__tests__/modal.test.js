import Modal from '../components/Modal/Modal.jsx';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../redux/store/index';
import renderer from 'react-test-renderer';
import { fireEvent, render, screen } from '@testing-library/react'


test('Modal component snapshot', () => {
    const modalData = [
        {
            header: 'Are you want to add this item in your card?',
        }
    ];
    const tree = renderer
        .create(
            <Provider store={store}>
                <BrowserRouter>
                    <Modal modalData={modalData} />
                </BrowserRouter>
            </Provider>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});

test('modal renders', () => {
    render(
        <Modal />
    )
    const modal = screen.getByTestId('modal')
    expect(modal).toBeInTheDocument()
})

test('modal closes', () => {
    const onClose = jest.fn()
    render(
        <Modal onClose={onClose} />
    )
    const overlay = screen.getByTestId("overlay");
    fireEvent.click(overlay);
    expect(onClose).toHaveBeenCalledTimes(1)
})


test('modal renders with modal data', () => {
    const modalDataMock = {
        header: 'Are you want to add this item in your card?',
        
      };
      render(
        <Modal modalData={modalDataMock} />
      );
    
      const modal = screen.getByTestId("modal");
      const title = screen.getByTestId("title");
      expect(modal).toBeInTheDocument();
      expect(title).toHaveTextContent(modalDataMock.header);
      
})