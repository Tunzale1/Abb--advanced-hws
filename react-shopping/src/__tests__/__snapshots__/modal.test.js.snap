// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal component snapshot 1`] = `
<div
  className="modal-overlay"
  data-testid="overlay"
>
  <div
    className="modal"
    data-testid="modal"
    onClick={[Function]}
  >
    <div
      className="modal-content"
    >
      <div
        className="modal-header"
      >
        <h3
          className="modal-title"
          data-testid="title"
        />
        <span
          className="close-btn"
        >
          <i
            className="fa-regular fa-2x fa-square-minus"
          />
        </span>
      </div>
      <div
        className="modal-footer"
      />
    </div>
  </div>
</div>
`;
